java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java

```diff
@@ -3731,6 +3746,11 @@ private void addTypes(Env env, EnumSet<ElementKind> kinds, DeclaredType baseType
             addLocalAndImportedTypes(env, kinds, baseType);
             hasAdditionalClasses = true;
         }
+        if (SOURCE_VERSION_RELEASE_10 != null &&
```

Comment:
Consider moving this to the `addKeywordsForBlock` method. Check for `env.getPath().getLeaf().getKind() == Tree.Kind.BLOCK` would not be necessary there. Also, check for `Utilities.startsWith(VAR_KEYWORD, prefix)` should be added.